### PR TITLE
Add env and env_file support to machine configuration

### PR DESCRIPTION
**Pull request by tuler**
## Motivation

Environment variables are typically used to differentiate between multiple build or test scenarios. One classic examples are testnet vs devnet vs mainnet, but there may be other scenarios.

Currently the build process only supports environment variables defined inside the Dockerfile used by the cartesi build process. But the same variables defined in the Dockerfile may be necessary outside the cartesi build process. For example in a host build process that is not related to the cartesi build process at all. So in this case there is a duplication of environment variables values, which hurts maintenance.

By having `.env` files support we can maintain a single source of truth for environment variables and use it in both the cartesi build workflow and other workflows.

## Summary
This PR adds support for injecting environment variables into the Cartesi Machine through two new configuration options in the `[machine]` section of `cartesi.toml`: an `env` table for inline key/value pairs and an `env_file` option pointing to a `.env` file.

## Key Changes
- **New configuration options**: Added `env` (Record<string, string>) and `envFile` (optional string) to `MachineConfig`
- **TOML parsing**: Implemented `parseStringRecord()` to parse TOML tables into environment variable records, with automatic coercion of non-string scalar values (numbers, booleans) to strings
- **`.env` file parsing**: Added `parseEnvFile()` function that parses `.env` files with support for:
  - Blank lines and `#` comments
  - Optional `export` prefix
  - Single/double quoted values
  - Whitespace trimming
  - Preservation of `=` signs in values
- **Environment variable precedence**: Updated `bootMachine()` to merge environment variables with proper precedence (lowest to highest):
  1. Docker image `ENV` (when `use_docker_env` is enabled)
  2. Variables from `env_file`
  3. Variables from the `env` table
- **Error handling**: Added `InvalidEnvError` exception class for invalid env configurations
- **Comprehensive tests**: Added unit tests covering all parsing scenarios and edge cases

## Implementation Details
- Environment variables are merged into a single map before being passed to `cartesi-machine`, allowing later sources to override earlier ones
- The `.env` file parser is robust and handles common `.env` file formats
- Non-string values in the `env` table are automatically converted to strings since environment variables are always strings
- The implementation maintains backward compatibility with existing configurations

https://claude.ai/code/session_016vnD51QAGwAvqaZuEZJypr

### CODE REVIEW

File: apps/cli/src/machine.ts [hunk 1]
```diff
@@ -42,12 +46,33 @@ export const bootMachine = (
...
+
+    // environment variables of docker image
+    if (useDockerEnv) {
+        for (const variable of info?.env ?? []) {
+            const eq = variable.indexOf("=");
+            if (eq === -1) {
+                envMap[variable] = "";
+            } else {
+                envMap[variable.slice(0, eq)] = variable.slice(eq + 1);
+            }
```
Comment (brunomenezes) [inline apps/cli/src/machine.ts]: @tuler,
These slices are odd. Could it be done by using a destructuring based on the assumptions of the `variable` content?

e.g.
```js
const [key, value=""] = variable.split("=");
envMap[key] = value;
```

**Comment by brunomenezes** [comment]
The PR description is strongly on the "how", but I don't understand clearly the "why".  Does it help solve an issue Lyno opened? 
* #477

**Comment by tuler** [comment]
> The PR description is strongly on the "how", but I don't understand clearly the "why".

Added a motivation section to the PR description.

File: apps/cli/tests/unit/config/fixtures/full.toml [hunk 2]
```diff
@@ -11,6 +11,11 @@
 # use_docker_env = true
 # use_docker_workdir = true
 # user = "dapp"
+# env_file = ".env" # optional. path to a .env file with environment variables
+
+# [machine.env] # optional. environment variables injected into the machine
+# FOO = "bar"
+# LOG_LEVEL = "debug"
```
Comment (brunomenezes) [inline apps/cli/tests/unit/config/fixtures/full.toml:18]: @tuler, 
I am curious and just trying to understand how to use it. Is the idea to plug in anything from the environment variable, like arbitrary values that only make sense to what is running inside, or is it to change the behaviour of the machine, like the `rollups_node` env vars?  If the latter would be good to have a list of behaviour-changing ones as they're specific and "reserved" in some sense.

PS: When I build dapps, the default configurations so far work for me, so I am trying to grasp the use. Even though you mentioned the **motivation** as the workflows and tests on different nets (dev, test ,main), it is not clear to me what these cases are; I mean what users are actually doing.

Comment (tuler) [reply apps/cli/tests/unit/config/fixtures/full.toml:18]: it's for any arbitrary env var the application may use to configure the application logic.
examples for a financial application:
COLLATERAL_TOKEN_ADDRESS
PRICE_FEED_ADDRESS

**Review by brunomenezes — APPROVED** [review]
